code/modules/mob/living/carbon/xenomorph/castes/hunter/hunter.dm

```diff
@@ -24,3 +24,13 @@
 
 /mob/living/carbon/xenomorph/hunter/weapon_x
 	caste_base_type = /mob/living/carbon/xenomorph/hunter/weapon_x
+
+/mob/living/carbon/xenomorph/hunter/weapon_x/Initialize(mapload)
+	. = ..()
+	RegisterSignal(SSdcs, COMSIG_GLOB_CAMPAIGN_MISSION_ENDED, PROC_REF(terminate_specimen))
```

Comment:
Make sure you clean up the signal in the Destroy proc

Reply:
are sigs not all cleaned up when destroyed anyway?

Reply:
Any reason not to just apply the signal on when the hunter is spawned via the xeno pod, or is there a reason to do it on the mob itself via a separate hunter type.

Reply:
what

Reply:
Ehh nevermind it wouldn't work anyways, sorry

Reply:
The top parent of destroys handles signals, you don't need to manually unregister.

https://github.com/tgstation/TerraGov-Marine-Corps/blob/949bc7a8a7bf4b69c67c8cf0d491053705b93733/code/datums/datum.dm#L125